crates/sui-indexer/src/handlers/pruner.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::time::Duration;
+
+use tokio_util::sync::CancellationToken;
+use tracing::{error, info};
+
+use crate::{metrics::IndexerMetrics, store::IndexerStore, types::IndexerResult};
+
+pub struct Pruner<S> {
+    pub store: S,
+    pub epochs_to_keep: u64,
+    pub metrics: IndexerMetrics,
+}
+
+impl<S> Pruner<S>
+where
+    S: IndexerStore + Clone + Sync + Send + 'static,
+{
+    pub fn new(store: S, epochs_to_keep: u64, metrics: IndexerMetrics) -> Self {
+        Self {
+            store,
+            epochs_to_keep,
+            metrics,
+        }
+    }
+
+    pub async fn start(&self, cancel: CancellationToken) -> IndexerResult<()> {
+        loop {
+            if cancel.is_cancelled() {
+                info!("Pruner task cancelled.");
+                return Ok(());
+            }
+
+            let (mut min_epoch, mut max_epoch) = self.store.get_available_epoch_range().await?;
+            while min_epoch + self.epochs_to_keep > max_epoch {
```

Comment:
what if we get a cancellation signal while in this loop? it seems like the pruner will get stuck here